python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -0,0 +1,1845 @@
+            epoch=epoch,
+            equinox=equinox,
+            parallax=parallax,
+            pmRA=pmRA,
+            pmDec=pmDec,
+            rv=rv,
+            dRA=dRA,
+            dDec=dDec,
+            rotFrame=rot_frame,
+            rotMode=rot_mode,
+        )
+
+        await self._slew_to(
+            self.rem.atptg.cmd_raDecTarget,
+            slew_timeout=slew_timeout,
+            stop_before_slew=stop_before_slew,
+            wait_settle=wait_settle,
+        )
+
+    async def slew_to_planet(self, planet, rot_pa=0.0, slew_timeout=1200.0):
+        """Slew and track a solar system body.
+
+        Parameters
+        ----------
+        planet : `ATPtg.Planets`
+            Enumeration with planet name.
+        rot_pa : `float`
+            Desired instrument position angle (degree), Eastwards from North.
+        slew_timeout : `float`
+            Timeout for the slew command (second).
+        """
+        self.rem.atptg.cmd_planetTarget.set(
+            planetName=planet.value, dRA=0.0, dDec=0.0, rotPA=rot_pa,
+        )
+
+        await self._slew_to(self.rem.atptg.cmd_planetTarget, slew_timeout=slew_timeout)
+
+    async def slew_dome_to(self, az):
+        """Utility method to slew dome to a specified position.
+
+        This method operates against ATDomeTrajectory, which means it can
+        only run if ATDomeTrajectory is disabled. As such, before moving the
+        dome it will set ATDomeTrajectory state to DISABLED and Fail if
+        ATDomeTrajectory transitions to ENABLED. In the end, it will leave
+        ATDomeTrajectory in DISABLED state, otherwise it will try to
+        synchronize the dome again.
+
+        Parameters
+        ----------
+        az : `float` or `str`
+            Azimuth angle for the dome (in deg).
+        """
+        await salobj.set_summary_state(self.rem.atdometrajectory, salobj.State.DISABLED)
+
+        self.rem.atdome.evt_azimuthInPosition.flush()
+
+        target_az = Angle(az, unit=u.deg).deg
+        await self.rem.atdome.cmd_moveAzimuth.set_start(
+            azimuth=target_az, timeout=self.long_long_timeout
+        )
+
+        self.check.atmcs = False
+        self.check.atdometrajectory = False
+        self.check.atdome = True
+
+        coro_list = [
+            asyncio.create_task(
+                self.wait_for_inposition(timeout=self.long_long_timeout)
+            ),
+            asyncio.create_task(self.monitor_position()),
+            asyncio.create_task(
+                self.check_component_state("atdometrajectory", salobj.State.DISABLED)
+            ),
+        ]
+
+        for res in asyncio.as_completed(coro_list):
+            try:
+                await res
+            except Exception:
+                raise
+            else:
+                break
+            finally:
+                await self.cancel_not_done(coro_list)
+                self.check.atmcs = True
+                self.check.atdometrajectory = True
```

Comment:
No, I don't think this will actually replace the code above.

```
await asyncio.gather(*coro_list)
```

will wait for all the tasks to complete before returning... I need to return as soon as any of them are done..

Reply:
Ah. That's fair. Given that you have repeated this same thing in several places and it's a bit complicated, please consider making it a subroutine and documenting what it does.